Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -72,81 +72,138 @@ namespace Babylon
             return static_cast<bgfx::TextureFormat::Enum>(format);
         }
 
-        void FlipY(bimg::ImageContainer* image)
+        using TransformFn = void (*)(const uint8_t*, uint8_t*);
+        void TransformImage(const bimg::ImageContainer* src, bimg::ImageContainer* dst, TransformFn transformFn)
         {
-            uint8_t* bytes = static_cast<uint8_t*>(image->m_data);
-            uint32_t rowCount = image->m_height;
-            uint32_t rowPitch = image->m_size / image->m_height;
+            const uint8_t* srcData{static_cast<const uint8_t*>(src->m_data)};
+            const uint8_t srcBytesPerPixel{static_cast<uint8_t>(bimg::getBitsPerPixel(src->m_format) >> 3)};
+            const uint8_t* srcDataEnd{srcData + src->m_size};
 
-            std::vector<uint8_t> buffer(rowPitch);
+            uint8_t* dstData{static_cast<uint8_t*>(dst->m_data)};
+            const uint8_t dstBytesPerPixel{static_cast<uint8_t>(bimg::getBitsPerPixel(dst->m_format) >> 3)};
+            const uint8_t* dstDataEnd{dstData + dst->m_size};
 
-            for (size_t row = 0; row < rowCount / 2; row++)
+            while (srcData < srcDataEnd && dstData < dstDataEnd)
```

Comment:
Moving from 3 memcpy per row to 1 call per pixel, how is the performance? I guess it will be way slower as memcpy are inlined and can use SIMD in release builds.

Reply:
This code moved. I didn't actually change FlipY which is now inside the PrepareImage function. The code here is a new function I added called TransformImage which does a transform per pixel.